src/commands/install.ts

```diff
@@ -30,43 +42,92 @@ export default class Install extends Command {
     }),
   };
 
-  private async installSingle(service_id: string) {
-    const service_dir = process.cwd();
-    const parts = service_id.split(':');
-    const service_name = parts[0];
-    let service_tag = parts[1];
-    const config = ServiceConfig.loadFromPath(service_dir);
+  async run() {
+    await this.tasks();
+  }
 
-    if (config.name === service_name) {
-      throw new Error('Services cannot depend on themselves');
-    }
+  async tasks() {
+    const { args, flags } = this.parse(Install);
+    const root_service_path = flags.prefix ? flags.prefix : process.cwd();
+    const dependency_graph = await generateGraphFromPaths([process.cwd()], new EnvironmentConfigV1(), this.app.api);
+    const root_service = Array.from(dependency_graph.nodes.values())[0];
 
-    const { data: service } = await this.app.api.get(`/services/${service_name}`);
-    if (!service) {
-      throw new Error(`Service not found in the registry`);
-    }
+    if (args.service_name) {
+      // eslint-disable-next-line prefer-const
+      let [service_name, service_version] = args.service_name.split(':');
+      if (!service_version) {
+        throw new Error(`Please specify a version for ${service_name}`);
+      }
+      if (root_service.name === service_name) {
+        throw new Error('Cannot install a service inside its own config');
+      }
+
+      const new_dependencies: { [s: string]: string } = {};
+      if (new_dependencies[service_name]) {
+        throw new Error(`A version of ${service_name} is already installed.`);
+      }
 
-    // Get the latest tag if none exists
-    if (!service_tag) {
-      service_tag = service.tags.sort((a: string, b: string) => b.localeCompare(a, undefined, { numeric: true }))[0];
+      cli.action.start(chalk.blue(`Installing ${args.service_name} as dependency of ${root_service.name}`), undefined, { stdout: true });
+      new_dependencies[service_name] = service_version;
+      const config = ServiceConfig.loadFromPath(root_service_path);
+      const all_dependencies = Object.assign({}, config.dependencies, new_dependencies);
+      config.setDependencies(all_dependencies);
+      const api_definitions_contents = await this.get_remote_definitions(args.service_name);
+      if (!api_definitions_contents) {
+        throw new Error(`No api definitions found for ${service_name}`);
+      }
+      await ProtocExecutor.execute((root_service as LocalServiceNode), undefined, { api_definitions_contents, service_name });
+      ServiceConfig.saveToPath(root_service_path, config);
+      cli.action.stop(chalk.green(`${args.service_name} installed`));
+    } else {
+      await this.installServices((root_service as LocalServiceNode), dependency_graph, flags.recursive);
     }
+  }
+
+  async get_remote_definitions(remote_service_version: string) {
+    const repository_name = url.resolve(`${this.app.config.registry_host}/`, `${remote_service_version}`);
```

Comment:
Replace with the following which will work better if we ever change repo naming conventions:

```ts
const { data: service } = await this.app.api.get(`/services/${service_name}`);
const repository_url = service.url.replace(/(^\w+:|^)\/\//, ''); // strips the protocol from the URL
const repository_name = `${repository_url}:${tag_name}`;
```